modules/gapi/src/streaming/onevpl/accelerators/surface/dx11_frame_adapter.cpp

```diff
@@ -0,0 +1,232 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2021 Intel Corporation
+
+#include "streaming/onevpl/accelerators/surface/dx11_frame_adapter.hpp"
+#include "streaming/onevpl/accelerators/dx11_alloc_resource.hpp"
+#include "streaming/onevpl/accelerators/surface/surface.hpp"
+#include "logger.hpp"
+
+#ifdef HAVE_ONEVPL
+#include "streaming/onevpl/onevpl_export.hpp"
+
+#ifdef HAVE_INF_ENGINE
+// For IE classes (ParamMap, etc)
+#include <inference_engine.hpp>
+#endif // HAVE_INF_ENGINE
+
+namespace cv {
+namespace gapi {
+namespace wip {
+namespace onevpl {
+
+void lock_mid(mfxMemId mid, mfxFrameData &data, MediaFrame::Access mode) {
+    LockAdapter* alloc_data = reinterpret_cast<LockAdapter *>(mid);
+    if (mode == MediaFrame::Access::R) {
+        alloc_data->read_lock(mid, data);
+    } else {
```

Comment:
since the new types of `Access` may be introduced in the future (e.g., I do consider `RW` to be the case), this `else` construct should be more robust I believe (or better to switch to `switch` already)

Reply:
ok, agree.
TODO in next PR